kaldb/src/test/java/com/slack/kaldb/blobfs/s3/S3BlobFsTest.java

```diff
@@ -27,32 +32,38 @@
   @ClassRule public static final S3MockRule S3_MOCK_RULE = S3MockRule.builder().silent().build();
 
   final String DELIMITER = "/";
-  static final String BUCKET = "test-bucket";
   final String SCHEME = "s3";
   final String FILE_FORMAT = "%s://%s/%s";
   final String DIR_FORMAT = "%s://%s";
 
   private final S3Client s3Client = S3_MOCK_RULE.createS3ClientV2();
+  private String bucket;
   private S3BlobFs s3BlobFs;
 
-  // TODO: Make this before and after class to speed up the unit test runs like Pinot tests.
-
   @Before
   public void setUp() {
+    bucket = "test-bucket-" + UUID.randomUUID();
```

Comment:
Test suite now uses a unique bucket per-test. Previous implementation resulted in shared state across tests, and when one test would fail it could cause the remainder of the tests to fail due to corrupted state.